codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java

```diff
@@ -102,6 +105,7 @@ public void close() { }
     @Override
     public ChannelFuture writeData(ChannelHandlerContext ctx, int streamId, ByteBuf data,
             int padding, boolean endStream, ChannelPromise promise) {
+        SimpleChannelPromiseAggregator promiseAggregator = null;
```

Comment:
Maybe just always create it?  It's not very expensive to create and it will almost never be non-null in the current flow.